deepchecks/vision/dataset.py

```diff
@@ -81,31 +80,33 @@ class VisionDataset:
     """
 
     _data: DataLoader = None
+    label_transformer: BaseLabelEncoder = None
 
     def __init__(self,
                  data_loader: DataLoader,
-                 label_type: str,
                  num_classes: Optional[int] = None,
-                 label_transformer: Optional[Callable] = None,
+                 label_transformer: Union[ClassificationLabelEncoder, DetectionLabelEncoder] = None,
                  sample_size: int = 1000,
                  sample_iteration_limit: int = 1_000_000,
                  random_seed: int = 0):
         self._data = data_loader
 
         if label_transformer is None:
-            self.label_transformer = lambda x: x
+            self.label_transformer = ClassificationLabelEncoder(lambda x: x)
         else:
             self.label_transformer = label_transformer
 
-        valid_label_types = [tt.value for tt in TaskType]
-        if label_type in valid_label_types:
-            self.task_type = TaskType(label_type)
+        if isinstance(self.label_transformer, ClassificationLabelEncoder):
+            self.task_type = TaskType.CLASSIFICATION
+        elif isinstance(self.label_transformer, DetectionLabelEncoder):
+            self.task_type = TaskType.OBJECT_DETECTION
         else:
-            raise DeepchecksValueError(f'Invalid label type: {label_type}, must be one of {valid_label_types}.')
+            raise ValueError(f'Unsupported label transformer type. {type(self.label_transformer)}, must be one of '
+                             f'ClassificationLabelEncoder or DetectionLabelEncoder.')
```

Comment:
Repeating my comment from Slack - why? We want to enable users to use task types that are not classification or object detection, we just won't have any default OOTB configurations for them. Why not enable them to use a general LabelEncoder?

Reply:
I can't see someone who will write a label encoder, prediction encoder, and metrics just to get a performance report.  In my opinion, the code is clearer that way. 
Most of the other checks won't work even with custom configuration as the assumptions of classification/regression are bound into the core logic of the checks.

Reply:
Tend to agree with @ItayGabbay on that

Reply:
Disagree. Many checks are agnostic to the task (all integrity)